Exygy/askdarcel-web
# Search Results section and related fixes

# Pull Request Template

## Issue Overview

This PR addresses [Search results for “” + Clear search](https://www.notion.so/exygy/Search-results-for-Clear-search-35e5d5f2a6fc4629a46b18d55b3ede9a)


## Description

Please include a summary of the change and which issue(s) is addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

This PR: 
- Adds the search results section on search pages per mockup. This shows only when there is a valid search query.

Other improvements:
- Improved the No Results message by adding a Clear Search button and including the query in the message
- Fixes buggy pagination which was only showing 1-20 on every page. Now it is chronological i.e. page 2 has 21-40.
- Updates the url to page=1 after the search query changes so users return to page 1 on fresh search

## How Can This Be Tested/Reviewed?

Visit /search and enter something into the search input. If there are results, there will be a section with the correct number of total results. If there are no results, there will be a message. Please check both on desktop and mobile.

Also, the ticket has some confusing language. The category pages and search page share the same components, but you can only filter NOT search the category page. Therefore, the search count is hidden on the category pages.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
